Log4jHotPatch.java

```diff
@@ -276,48 +432,63 @@ private static String getUID(String pid) {
     }
   }
 
+  /**
+   * Entrypoint into this Log4JHotPatch utility.
+   *
+   * @param args - Log4jHotPatch [<pid> [<pid> ..]]"
+   * @throws Exception - Note this program can crash fairly easily so make
+   *                     sure you are able to capture stderr
+   */
   public static void main(String args[]) throws Exception {
 
-    String pid[];
+    String jvmPidsToPatch[];
     if (args.length == 0) {
       MonitoredHost host = MonitoredHost.getMonitoredHost((String)null);
-      Set<Integer> pids = host.activeVms();
-      pid = new String[pids.size()];
+      Set<Integer> activeVms = host.activeVms();
```

Comment:
This approach to discover activeVMs is subject to PerfDataSampling on Linux.  ie.  it can not discover the JVMs which use `-XX:+PerfDisableSharedMem`(default value is false).

OTOH, explicitly specifying pid in the 1st argument still works.  Maybe we should show a warning message.

Reply:
@navyxliu so maybe you're looking for something like this?

```
try {
            MonitoredHost host = MonitoredHost.getMonitoredHost((String) null);
            Set<Integer> activeVmPids = host.activeVms();
            jvmPidsToPatch = new String[activeVmPids.size()];
}
 catch (URISyntaxException use) {
  // Can never be thrown as we are passing in null, a valid URI (weirdly enough)
}
 catch (MonitorException me) {
  logError("Unable to connect to the host: " + me + " Perhaps you have -XX:+PerfDisableSharedMem set to true.");
}
```

Reply:
I don't think you can catch the exception `MonitorException` there. 

The program will simply ignore those java processes which are using `-XX:+PerfDisableSharedMem`.
`jcmd` uses the same discovery mechanism. you will see that it can't discover those JVMs hosts.

I'd say we print a warning message to let users aware  it will leave those JVMs with -XX:+PerfDisableSharedMem` alone if `args.length == 0`

Reply:
OK I've added logging for this.